lib/phoenix_html/form.ex

```diff
@@ -515,6 +515,59 @@ defmodule Phoenix.HTML.Form do
     generic_input(:range, form, field, opts)
   end
 
+  @doc """
+  Generates a date input.
+
+  Warning: this feature isn't available in all browsers.
+  Check `http://caniuse.com/#feat=input-datetime` for further informations.
+
+  See `text_input/3` for example and docs.
+  """
+  def date_input(form, field, opts \\ []) do
+    generic_input(:date, form, field, opts)
+  end
+
+  @doc """
+  Generates a datetime-local input.
+
+  Warning: this feature isn't available in all browsers.
+  Check `http://caniuse.com/#feat=input-datetime` for further informations.
+
+  See `text_input/3` for example and docs.
+  """
+  def datetime_local_input(form, field, opts \\ []) do
+    opts = Keyword.replace(opts, :value, datetime_local_input_value(Keyword.get(opts, :value)))
+    generic_input(:'datetime-local', form, field, opts)
+  end
+
+  defp datetime_local_input_value(%{__struct__: NaiveDateTime} = value) do
+    str = to_string(value)
+    String.slice(str, 0..9) <> "T" <> String.slice(str, 11..15)
```

Comment:
This function could likely be more efficiently implemented as:

```elixir
<<date::10-binary, ?\s, hour_minute::5-binary, _rest::binary>> = NaiveDateTime.to_string(value)
[date, ?T, hour_minute]
```

Reply:
OK, will do!